src/cmd-cloud-prune

```diff
@@ -0,0 +1,303 @@
+
+        return json_content
+
+    except Exception as e:
+        print(f"Error fetching the JSON file from S3 {bucket}/{key}: {e}")
+        return None
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(prog="coreos-assembler cloud-prune")
+    parser.add_argument("--policy", default='./policy.yaml', type=str,
+                        help="Path to policy.yaml file")
+    parser.add_argument("--dry-run", help="Don't actually delete anything",
+                        action='store_true')
+    parser.add_argument("--stream", type=str, help="Fedora stream", required=True)
+
+    parser.add_argument("--gcp-json-key", help="GCP Service Account JSON Auth",
+                        default=os.environ.get("GCP_JSON_AUTH"))
+    parser.add_argument("--gcp-project", help="GCP Project name",
+                        default=os.environ.get("GCP_PROJECT_NAME"))
+
+    subparsers = parser.add_subparsers(dest="cmd", title="subcommands")
+    subparsers.required = True
+
+    s3 = subparsers.add_parser('s3', help='Prune s3 buckets')
+    s3.add_argument("url", metavar='<BUCKET>[/PREFIX]',
+                    help="Bucket and path prefix in which to upload")
+    s3.add_argument("--acl", help="ACL for objects",
+                    action='store', default='private')
+    s3.add_argument("--aws-config-file", default=os.environ.get("AWS_CONFIG_FILE"),
+                    help="Path to AWS config file")
+    return parser.parse_args()
+
+
+# Handling just AWS and GCP at the moment
+def delete_cloud_resources(build, gcp_cloud_config, dry_run):
+    errors = []
+    totalBuildAMIs = len(build.images.get("amis") or "")
+    GCPImage = 0
+    # Unregister AMIs and snapshots
+    if not build.images.get("amis", []):
+        print(f"No AMIs for {build.id} for {build.arch}")
+    if not build.images.get("gcp", []):
+        print(f"No GCP image for {build.id} for {build.arch}")
+    else:
+        GCPImage = 1
+
+    if dry_run:
+        print(f"Would delete {totalBuildAMIs} amis/snapshots and {GCPImage} GCP image for {build.id} for {build.arch}")
+        return
+
+    for ami in build.images.get("amis", []):
+        print(f"Deleting cloud uploads for {build.id}")
+        region_name = ami.get("name")
+        ami_id = ami.get("hvm")
+        snapshot_id = ami.get("snapshot")
+        if ami_id and region_name:
+            try:
+                deregister_ami(ami_id, region=region_name)
+            except Exception as e:
+                errors.append(e)
+        if snapshot_id and region_name:
+            try:
+                delete_snapshot(snapshot_id, region=region_name)
+            except Exception as e:
+                errors.append(e)
+
+    gcp = build.images.get("gcp")
+    if gcp:
+        gcp_image = gcp.get("image")
+        json_key = gcp_cloud_config.get("gcp", {}).get("json-key")
+        project = gcp_cloud_config.get("gcp", {}).get("project")
+        if gcp_image and json_key and project:
```

Comment:
I feel like we should error out here if the information is incomplete.

Reply:
Do you mean in the case of either of the parameters (`gcp_image` and `json_key` and `project`) not being present.

Reply:
Yes. We can't add the `cloud-uploads` marker to a build if we couldn't actually remove a cloud image just because we were missing information about how to remove it.